src/kwlib.sh

```diff
@@ -418,3 +422,88 @@ function kw_parse_get_errors()
       -- "$@" > /dev/null
   } 2>&1
 }
+
+# This function compresses a given path to a .tar.gz file
+#
+# @path_to_compress The directory to compress
+# @file_path Where the compressed directory will be stored
+# @compression_type compression program used
+# @flag How to display (or not) the command used
+function generate_tarball()
+{
+  local path_to_compress="$1"
+  local file_path="$2"
+  local compression_type="$3"
+  local flag="$4"
+  local ret
+  local cmd
+
+  flag=${flag:-'SILENT'}
+
+  if [[ ! -d "$path_to_compress" ]]; then
+    complain "$path_to_compress" 'does not exist'
+    exit 22 #EINVAL
+  fi
+
+  if [[ -n "$compression_type" ]]; then
+    if [[ "${compression_programs[*]}" =~ $compression_type ]]; then
+      compression_type="--$compression_type"
+    else
+      complain 'Invalid compression type:' "$compression_type"
+      return 22 # EINVAL
+    fi
+  fi
+
+  compression_type=${compression_type:-'--auto-compress'}
+
+  cmd="tar -C $path_to_compress $compression_type -cf $file_path ."
+  cmd_manager "$flag" "$cmd"
+
+  ret="$?"
+
+  if [[ "$ret" != 0 ]]; then
+    complain "Error archiving modules."
+    exit "$ret"
+  fi
```

Comment:
It looks like that we are not testing this path. Could you add a test for that? Also, use single quotes.